src/blob/index/bptree/node.rs

```diff
@@ -0,0 +1,117 @@
+use super::prelude::*;
+
+#[derive(Debug, PartialEq)]
+pub(super) struct Node {
+    keys: Vec<Vec<u8>>,
+    offsets: Vec<u64>,
+}
+
+impl Node {
+    pub(super) fn new(keys: Vec<Vec<u8>>, offsets: Vec<u64>) -> Self {
+        assert!(keys.len() != 0);
+        assert_eq!(keys.len() + 1, offsets.len());
+        Self { keys, offsets }
+    }
+
+    pub(super) fn key_offset(&self, key: &[u8]) -> u64 {
+        let mut left = 0i32;
```

Comment:
You can use "self.data.binary_search_by". It returns either found elem or position at which elem can be inserted (which can be used in case nothing is found to return offset)